src/main/java/com/trueffect/validation/PersonValidation.java

```diff
@@ -1,75 +1,105 @@
-                   Pattern.matches(RegularExpression.NIT,identifier) ;
+
+    public static boolean isValidTypeIdentifier(String typeId) {
+        boolean res = true;
+        try {
+            TypeIdentifier typeIdentifier = TypeIdentifier.valueOf(typeId);
+        } catch (Exception e) {
+            res = false;
         }
-        return false;
+        return res;
     }
 
-    public static boolean isValidFirstName(String firstName, int size){
-         if(firstName.length()>0) 
-    return firstName.length()<= DataResourse.MAXIMUM_VALUES&&Pattern.matches(RegularExpression.FIRST_NAME,firstName);
-         return false;
+    public static boolean isValidIdentifier(String identifier) {
+        return Pattern.matches(RegularExpression.CI, identifier)
+                || Pattern.matches(RegularExpression.PASS, identifier)
+                || Pattern.matches(RegularExpression.NIT, identifier);
+
     }
-    public  static boolean isValidSize(String name, int size){
-     return name.length()>=0 && name.length()<= DataResourse.MAXIMUM_VALUES;
+
+    public static boolean isValidFirstName(String firstName) {
+        return Pattern.matches(RegularExpression.FIRST_NAME, firstName);
     }
-    
-   public static boolean isValidLastName(String lastName, int size){
-    if(lastName.length()>0) 
-        if(lastName.contains(" "))
-       return lastName.length()<= DataResourse.MAXIMUM_VALUES&&Pattern.matches(RegularExpression.LAST_NAME_TWO,lastName);
-        
-    return lastName.length()<= DataResourse.MAXIMUM_VALUES&&Pattern.matches(RegularExpression.LAST_NAME_ONE,lastName);
+
+    public static boolean isValidSize(String name, int size) {
+        return name.length() <= size;
     }
-  
-    public static boolean isValidGenre(String genre) {
-        Genre genreEnum= null;
-        if (genre != null) {
-            return genre.equals(genreEnum.M.name()) || genre.equals(genreEnum.F.name());
+
+    public static boolean isValidLastName(String lastName) {
+        if (lastName.contains(" ")) {
+            return Pattern.matches(RegularExpression.LAST_NAME_TWO, lastName);
+        } else {
+            return Pattern.matches(RegularExpression.LAST_NAME_ONE, lastName);
         }
-        return false;
+    }
+
+    public static boolean isValidGenre(String genre) {
+        Genre genreEnum = null;
+        return genre.equals(genreEnum.M.name()) || genre.equals(genreEnum.F.name());
     }
 
     public static boolean isValidBirthday(String date) {
-        if (date.length() > 0) {
-            return Pattern.matches(RegularExpression.DATE, date);
-        }
-        return false;
+        return Pattern.matches(RegularExpression.DATE, date);
     }
-   
 
-    public static boolean isValidIdentifier(String typeIdentifier, String identifier)  throws Exception{
-     return true;
+    public static boolean isValidAge(String age) {
+        boolean res = false;
+        Date date = new Date();
```

Comment:
Please use the library of java  get the year and recibe as parameter the age to validate